src/Forms/GridField/GridFieldEditButton.php

```diff
@@ -95,14 +106,63 @@ public function getColumnContent($gridField, $record, $columnName)
         // which can make the form readonly if no edit permissions are available.
 
         $data = new ArrayData(array(
-            'Link' => Controller::join_links($gridField->Link('item'), $record->ID, 'edit')
+            'Link' => Controller::join_links($gridField->Link('item'), $record->ID, 'edit'),
+            'ExtraClass' => $this->getExtraClass()
         ));
 
         $template = SSViewer::get_templates_by_class($this, '', __CLASS__);
         return $data->renderWith($template);
     }
 
     /**
+     * Get the extra HTML classes to add for edit buttons
+     *
+     * @return string
+     */
+    public function getExtraClass()
+    {
+        return implode(' ', Convert::raw2att($this->extraClass));
```

Comment:
The `default_cast` should have already cast these as safely escaped text. The convert is not needed here.